polkadot/runtime/parachains/src/paras/mod.rs

```diff
@@ -1230,6 +1234,10 @@ impl<T: Config> Pallet<T> {
 		// Check that we can schedule an upgrade at all.
 		ensure!(Self::can_upgrade_validation_code(id), Error::<T>::CannotUpgradeCode);
 		let config = configuration::Pallet::<T>::config();
+		// Validation code sanity checks:
+		ensure!(new_code.0.len() > 0, Error::<T>::EmptyCode);
```

Comment:
A compressed blob should contain at least 8 bytes of ZSTD signature and 1 byte of payload, thus 9 bytes.

If it's uncompressed, it should contain a 4-byte Wasm magic, a 4-byte version, and at least one section.

Thus, it's safe to assume the minimal possible length of valid code is 9 bytes.

But I'm also okay with keeping this a more general "just-non-zero" check.

Reply:
Such assumptions should then probably be checked by the `ValidationCode` type itself, maybe with some `check_sanity` function.  
Accessing the `.0` here and making assumptions about it seems unclean to me.

Reply:
Fair enough! I meant just some stricter checks, but from the design point of view, you have totally valid point here.